spdm.rflx

```diff
@@ -0,0 +1,1487 @@
+         Reserved_4 : Reserved_16;
+         Ext_Asym : Ext_Algs
+            with Size => 32 * Ext_Asym_Count;
+         Ext_Hash : Ext_Algs
+            with Size => 32 * Ext_Hash_Count
+            then Resp_Alg_Structs
+               --  Length is the total length of the message, we have to subtract the common header
+               --  of Request.
+               with Size => 8 * (Length_2 * 256 + Length_1) - 16 - (Ext_Hash'Last - Resp_Alg_Struct_Count'First + 1)
+                  if 8 * (Length_2 * 256 + Length_1) >= 16 + (Ext_Hash'Last - Resp_Alg_Struct_Count'First + 1);
+         Resp_Alg_Structs : Resp_Alg_Structs;
+         -- ISSUE: Componolit/systematization-binary-schemas#8
+         -- The number of elements should also match Resp_Alg_Struct_Count
+      end message;
+
+   -- 10.7 GET_DIGESTS request and DIGESTS response messages
+
+   type Get_Digests_Request is
+      message
+         Param_1 : Reserved_8;
+         Param_2 : Reserved_8;
+      end message;
+
+   type Digests_Response is
+      message
+         Param_1 : Reserved_8;
+         Slot_Mask : Slot_Mask;
+         Digests_Data : Hashes
+            --  FIXME: Hash'Size should work, but doesn't
+            with Size => 8 * 48 * (Slot_Mask_Slot_0_Present
+                                   + Slot_Mask_Slot_1_Present
+                                   + Slot_Mask_Slot_2_Present
+                                   + Slot_Mask_Slot_3_Present
+                                   + Slot_Mask_Slot_4_Present
+                                   + Slot_Mask_Slot_5_Present
+                                   + Slot_Mask_Slot_6_Present
+                                   + Slot_Mask_Slot_7_Present);
+      end message;
+
+   -- 10.8 GET_CERTIFICATE request and CERTIFICATE response messages
+
+   type Slot is
+      (Slot_0              => 16#00#,
+       Slot_1              => 16#01#,
+       Slot_2              => 16#02#,
+       Slot_3              => 16#03#,
+       Slot_4              => 16#04#,
+       Slot_5              => 16#05#,
+       Slot_6              => 16#06#,
+       Slot_7              => 16#07#,
+       Trusted_Environment => 16#FF#)
+   with Size => 8;
+
+   type Offset is mod 2**16;
+
+   type Get_Certificate_Request is
+      message
+         Slot : Slot;
+         Param_2 : Reserved_8;
+         --  This property must be expressed in the session:
+         --     For the first GET_CERTIFICATE request, the Requester shall set this field to 0.
+         --     For non-first requests, Offset is the sum of PortionLength values in all previous
+         --     GET_CERTIFICATE responses.
+         Offset : Offset;
+         Length_1 : Length;
+         Length_2 : Length;
+      end message;
+
+   --  ISSUE: Componolit/RecordFlux#609
+   --  This message differs between SPDM version 1.0 and 1.1.
+   --  Since the minor version is not accessible from
+   --  within this message one message is needed for each version
+   --  that is selected by a refinement.
```

Comment:
We should update this comment, we don't use refinements anymore.

Reply:
Componolit/RecordFlux#609 is fixed. This actually is part of #92 where the duplicate message should be merged again.

Reply:
Fixed in https://github.com/Componolit/RecordFlux-specifications/pull/93/commits/7c1908a0db7eece9cbbb987fa14b0c9d9e4ec91f.